File: exercise/python/day10/delivery/building.py
class Building:
    @staticmethod
    def which_floor(instructions: str) -> int:
        val_list = []
        result = 0

        for i in range(len(instructions)):
            c = instructions[i]

            if "🧝" in instructions:
                j = 3 if c == ')' else -2
                val_list.append((c, j))
            elif "🧝" not in instructions:
                val_list.append((c, 1 if c == '(' else -1))
            else:
                val_list.append((c, 42 if c == '(' else -2))

        for kp in val_list:
            result += kp[1]

        return result

File: exercise/ts/day10/src/delivery/building.ts
export class Building {
    static whichFloor(instructions: string): number {
        let val: Array<[string, number]> = [];

        for (let i = 0; i < instructions.length; i++) {
            const c = instructions[i];

            if (instructions.includes("🧝")) {
                const j = c === ')' ? 3 : -2;
                val.push([c, j]);
            } else if (!instructions.includes("🧝")) {
                val.push([c, c === '(' ? 1 : -1]);
            } else {
                val.push([c, c === '(' ? 42 : -2]);
            }
        }

        let result = 0;
        for (const kp of val) {
            result += kp[1];
        }

        return result;
    }
}